Src/BootCamp.Chapter/Lesson4.cs

```diff
@@ -0,0 +1,117 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace BootCamp.Chapter
+{
+    class Lesson4
+    {
+        public static void Demo()
+        {
+            string name = PromptString("What's your first name?");
+            string surname = PromptString("What's your surname?");
+            int age = PromptInt("How old are you?");
+            float weight = PromptFloat("How much do you weight in kg?");
+            float height = PromptFloat("How tall are you in cm?");
+            float mHeight = ConvertToMetres(height);
+            float bodyMassIndex = CalculateBMI(mHeight, weight);
+
+            Console.WriteLine(name + " " + surname + " is " + age + " years old, his weight is " + weight + "kg and his height is " + height + " cm");
+
+            Console.WriteLine("Their BMI is: " + bodyMassIndex);
+
+            string friendName = PromptString("\nWhat's your friend's first name?");
+            string friendSurname = PromptString("What's their surname?");
+            int friendAge = PromptInt("How old are they?");
+            float friendWeight = PromptFloat("How much do they weight in kg?");
+            float friendHeight = PromptFloat("How tall are they in cm?");
+            float friendMHeight = ConvertToMetres(friendHeight);
+            float friendBodyMassIndex = CalculateBMI(friendMHeight, friendWeight);
+
+            Console.WriteLine(friendName + " " + friendSurname + " is " + friendAge + " years old, his weight is " + friendWeight + "kg and his height is " + friendHeight + " cm");
+
+            Console.WriteLine("Their BMI is: " + friendBodyMassIndex);
+        }
+
+        public static string PromptString(string message)
+        {
+            Console.WriteLine(message);
+            string response = Console.ReadLine();
+
+            if (string.IsNullOrEmpty(response))
```

Comment:
In programming there is a principle to "Fail Early". Or "Finish early", "Exit early". This means, that every function should start with validation and return the result (or error) as soon as it can. Normally, validation is the easiest/quick way of getting the result, therefore you first validate stuff, return error/value if it was bad and only after that do the actual calculation. This makes us care about less things in a function and everything follows a similar pattern (more consistent)